cmdstanpy/model.py

```diff
@@ -601,20 +606,56 @@ def sample(
                             'Chain_id must be a non-negative integer value,'
                             ' found {}.'.format(chain_id)
                         )
-
-        cores_avail = cpu_count()
-        if cores is None:
-            cores = max(min(cores_avail - 2, chains), 1)
-        if cores < 1:
+        if parallel_chains is None or parallel_chains > chains:
+            parallel_chains = max(min(cpu_count(), chains), 1)
+        if parallel_chains > cpu_count():
+            self._logger.warning(
+                'Requested %u parallel chains, but only %u CPUs,'
+                ' can only run %u chains in parallel.',
+                threads_per_chain, cpu_count(), cpu_count()
+            )
+            parallel_chains = max(min(cpu_count(), chains), 1)
+        if parallel_chains < 1:
```

Comment:
This is probably following CmdStanR logic. In R, it is common to set `-1` == max

But this is fine.

Reply:
not sure I understand the comment about `-1 == max` ? as a test?  in any case, no longer doing this computation.